ibis/pandas/execution/window.py

```diff
@@ -130,24 +132,64 @@ def get_aggcontext_window(
     return aggcontext
 
 
+def trim_with_timecontext(data, timecontext):
+    try:
+        begin, end = map(pd.to_datetime, timecontext)
+    except ValueError:
+        raise com.IbisError('Cannot resolve timecontext for window\n{}.')
+    else:
+        df = data.reset_index()
+        time_col = 'time'
+        subset = df.loc[df[time_col].between(begin, end)]
+        name = data.name
+        non_target_columns = list(subset.columns.difference([name]))
+        indexed_subset = subset.set_index(non_target_columns)
+        return indexed_subset[name]
+
+
 @execute_node.register(ops.WindowOp, pd.Series, win.Window)
 def execute_window_op(
-    op, data, window, scope=None, aggcontext=None, clients=None, **kwargs
+    op,
+    data,
+    window,
+    scope=None,
+    timecontext=None,
+    aggcontext=None,
+    clients=None,
+    **kwargs,
 ):
     operand = op.expr
     # pre execute "manually" here because otherwise we wouldn't pickup
     # relevant scope changes from the child operand since we're managing
     # execution of that by hand
     operand_op = operand.op()
+
+    computable_args = [arg for arg in op.inputs if is_computable_input(arg)]
+    arg_timecontexts = compute_time_context(
+        op, computable_args, timecontext=timecontext
+    )
+
+    if len(arg_timecontexts):
```

Comment:
This if/else is a little strange. Can you explain what it is doing?

Reply:
It is removed and replaced with a check to timecontext, also docs to explain why